### PR TITLE
feat: add sentiment-analysis function example

### PR DESCRIPTION
**Pull request by pieterbrink123**
- Add new sentiment-analysis function based on auto-tag example
- Analyzes user-generated content sentiment with 5-level classification
- Supports review, comment, and feedback document types
- Includes comprehensive documentation and testing examples
- Enables automated workflow triggers based on sentiment detection

### What to review

Can someone test the recipe end-to-end?

### CODE REVIEW

File: examples/functions/sentiment-analysis/README.md [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+- **Use real document IDs** - Document functions require IDs that exist in your dataset
+- **Query for test documents** - Use `npx sanity documents query` to find suitable test documents
+- **Use Node.js v22.x** locally to match production runtime
+- **Test edge cases** like content without text or with existing sentiment
+- **Check function logs** in CLI output for debugging
+- **Test without AI calls** first by setting `noWrite: true` in the function
+- **Create test content** - If you don't have reviews without sentiment, create some test documents first
+- **Test various sentiment levels** - Try positive, negative, and neutral content
+
+## Requirements
```
Comment (markmichon) [inline examples/functions/sentiment-analysis/README.md]: I got caught by a few of these prior to reading this. Requirements might be more helpful before the code.

File: examples/functions/sentiment-analysis/README.md [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+
+- A Sanity project with Functions enabled
+- A schema with user-generated content document types containing:
+  - A `content` field with portable text (for content analysis)
+  - A `sentiment` field (for storing analyzed sentiment)
+- Access to Sanity's AI capabilities
```
Comment (markmichon) [inline examples/functions/sentiment-analysis/README.md]: While `_.schemas.default` should be safe for most users, it's worth adding a note either here or in the code comments telling them to update to their schema ID (from sanity schema list).

File: examples/functions/sentiment-analysis/README.md [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+        {
+          "_type": "span",
+          "_key": "test-span",
+          "text": "This product is absolutely terrible. I hate it and want my money back!"
+        }
+      ]
+    }
+  ]
+}
+EOF
```
Comment (markmichon) [inline examples/functions/sentiment-analysis/README.md]: I'm fairly sure `generate` ignores this data and will use the document ID (along with the instruction params) to pull data from the supplied ID. The test data is only used by the function, but not generate, so the only thing needed from this is an _id for generate to use and _type for the function filter to clear.

File: examples/functions/sentiment-analysis/README.md [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+
+```bash
+# From the studio/ folder, find and export a document that matches your function's filter
+cd studio
+npx sanity documents query "*[_type == 'review' && !defined(sentiment)][0]" > ../test-real-document.json
+
+# Back to project root for function testing
+cd ..
+npx sanity functions test sentiment-analysis --file test-real-document.json
+```
```
Comment (markmichon) [inline examples/functions/sentiment-analysis/README.md]: As of 3.99.0*, you can now do
`npx sanity functions test sentiment-analysis --document-id 13246 --dataset production`
to pull in the data from a real document.

*the release breaks `generate` with a bug, but if this goes out after that's fixed, will be much easier.

File: examples/functions/sentiment-analysis/index.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        noWrite: true,
+        instructionParams: {
+          content: {
+            type: 'field',
+            path: 'content',
+          },
+        },
+        instruction: `Analyze the sentiment of the $content and categorize it into one of these 5 levels: very_positive, positive, neutral, negative, very_negative. Consider the emotional tone, word choice, and overall sentiment expressed in the text. Return only the sentiment level as a string.`,
+        target: {
+          path: 'sentiment',
```
Comment (markmichon) [inline examples/functions/sentiment-analysis/index.ts]: I'd add comments in areas that need user intervention, like confirming `sentiment` is relative to the root of the document and not nested somewhere.

File: examples/functions/sentiment-analysis/README.md (hunk 4 above)
Comment (pelikenjones) [reply examples/functions/sentiment-analysis/README.md]: Ohhhh... This makes sense why generate stopped working halfway through testing 👀

File: examples/functions/sentiment-analysis/README.md [hunk 6]
```diff
@@ -0,0 +1,365 @@
...
+Start the development server for interactive testing:
+
+```bash
+npx sanity functions dev
+```
+
+This opens an interactive playground where you can test functions with custom data
+
+
+### 5. Enable Debugging
```
Comment (iamkevingreen) [inline examples/functions/sentiment-analysis/README.md]: @kenjonespizza This number is off now

**Comment by pelikenjones** [comment]
Closing.  Has been reopened with my permissions here -> #10106
